examples/all-clusters-app/all-clusters-common/include/DeviceEnergyManagementDelegateImpl.h

```diff
@@ -0,0 +1,92 @@
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "app/clusters/device-energy-management-server/device-energy-management-server.h"
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/MessageDef/StatusIB.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/CHIPError.h>
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace DeviceEnergyManagement {
+
+/** The application delegate.
+ */
+class DeviceEnergyManagementDelegate : public Delegate
+{
+public:
+    DeviceEnergyManagementDelegate();
+    virtual ~DeviceEnergyManagementDelegate() = default;
+
+    virtual Status PowerAdjustRequest(const int64_t power, const uint32_t duration) override;
+    virtual Status CancelPowerAdjustRequest() override;
+    virtual Status StartTimeAdjustRequest(const uint32_t requestedStartTime) override;
+    virtual Status PauseRequest(const uint32_t duration) override;
+    virtual Status ResumeRequest() override;
+    virtual Status
+    ModifyForecastRequest(const uint32_t forecastId,
+                          const DataModel::DecodableList<Structs::SlotAdjustmentStruct::DecodableType> & slotAdjustments) override;
+    virtual Status RequestConstraintBasedForecast(
+        const DataModel::DecodableList<Structs::ConstraintsStruct::DecodableType> & constraints) override;
+
+    // ------------------------------------------------------------------
+    // Get attribute methods
+    virtual ESATypeEnum GetESAType() override;
+    virtual bool GetESACanGenerate() override;
+    virtual ESAStateEnum GetESAState() override;
+    virtual int64_t GetAbsMinPower() override;
+    virtual int64_t GetAbsMaxPower() override;
+    virtual DataModel::Nullable<PowerAdjustmentCapability::TypeInfo::Type> GetPowerAdjustmentCapability() override;
+    virtual DataModel::Nullable<Structs::ForecastStruct::Type> GetForecast() override;
+
+    // ------------------------------------------------------------------
+    // Set attribute methods
+    virtual CHIP_ERROR SetESAType(ESATypeEnum) override;
+    virtual CHIP_ERROR SetESACanGenerate(bool) override;
+    virtual CHIP_ERROR SetESAState(ESAStateEnum) override;
+    virtual CHIP_ERROR SetAbsMinPower(int64_t) override;
+    virtual CHIP_ERROR SetAbsMaxPower(int64_t) override;
+    virtual CHIP_ERROR SetPowerAdjustmentCapability(Attributes::PowerAdjustmentCapability::TypeInfo::Type &) override;
+    virtual CHIP_ERROR SetForecast(Structs::ForecastStruct::Type &) override;
+
+private:
+    ESATypeEnum mEsaType;
+    bool mEsaCanGenerate;
+    ESAStateEnum mEsaState;
+    int64_t mAbsMinPower;
+    int64_t mAbsMaxPower;
+    DataModel::Nullable<PowerAdjustmentCapability::TypeInfo::Type> mPowerAdjustmentCapability;
```

Comment:
The underlying type (the template arg) is already nullable. 
I think you are adding another layer of nullability ? 
See below from :  zzz_generated/app-common/app-common/zap-generated/cluster-objects.h

struct TypeInfo
    using Type = chip::app::DataModel::Nullable< chip::app::DataModel::List<const
                                                      chip::app::Clusters::DeviceEnergyManagement::Structs::PowerAdjustStruct::Type>>;

Reply:
Fixed